### DISCUSSION TITLE
ミッション生成や確定を勝手にいじれてしまう件について

### DISCUSSION THREAD

**Discussion by masaki-tsuchikura** [masaki-tsuchikura/sleep-comm-ai#97] [Ideas]
私の意見としてはPriveteグループはルームを作成した人にミッション生成と確定の権限を与える。
Publicはチームメンバー確定の期限をもうけて、そのあと投票でリーダーを決め権限を与える。

Publicの場合何らかの方法でチームメンバーの確定がないとミッション開始2日目とかでPublicグループに途中で入ってくるみたいな問題も起きそうです。

**Comment by batabata-256** [comment]
私が懸念しているのは、極端に言えば「あああああ」のような無茶苦茶なミッションに書き換えられてしまうことです。これはユーザーにミッションを書き換えられるシステムにするのであれば、避けられない問題だと思います。

誰か1人のみにミッションの生成・確定権限を与えると、他の人の意向が反映されなくなってしまい、本来の「グループ内の各メンバーの特性を踏まえてミッション設定する」という目的が達成できなくなるように思えます。

海晴に再度修正を強いることになってしまいますが、私が考える代案は「グループの人数分のミッションを各々のトピックに基づいて生成し、その中から1つを多数決で決める」というものです。たとえば3人グループの場合、
| 名前 | トピック                     | ミッション                       |
|------|------------------------------|----------------------------|
| Aさん | 8時間寝たい                  | 8時間の睡眠時間を確保しよう |
| Bさん | 寝る前にスマホを見ない      | デジタルデトックスに取り組もう |
| Cさん | 頭がモヤモヤして眠れない    | 筆記開示をやってみよう       |

上記3つのミッションを生成・提示し、ユーザーに多数決を取らせるといった感じです。

生成するミッション数は必ずしもグループの人数分である必要はないですが、いずれにせよ「ミッションを書き換えられるようにするか否か」については、もう少し議論が必要そうです。

**Comment by batabata-256** [comment]
> Publicの場合何らかの方法でチームメンバーの確定がないとミッション開始2日目とかでPublicグループに途中で入ってくるみたいな問題も起きそうです。

そうですね。
グループ作成から6時間後に新規加入を打ち切り -> そこから6時間以内にミッションを決定
といった具合に調整が必要そうです。

**Comment by masaki-tsuchikura** [comment]
Publicグループだと全員抜けてもメッセージやミッションなどの情報が残ってしまう問題もあります。

**Reply by masaki-tsuchikura** [reply]
Publicグループがあると最後の発表に間に合わない気がするのでなくてもいいかなと思います。
Privateだとチームを作った人にミッション生成ボタンと確定ボタンの権限を渡すだけでいいので楽かなと思います。

メンバーの締切もいらないですし。ミッション生成ボタンの上に[メンバー全員入ったらこのボタンを押してください]みたいな説明を書けば大丈夫かなと思います。

**Comment by s1f10220256** [comment]
確かにプライベートのみで権限渡す方式なら、問題なさそうですね。
生成系AIを使うメリットを残すためにも、いくつかの選択肢提示して選択式が良さそうだなと思いました

**Comment by masaki-tsuchikura** [comment]
メンバー締め切るの難しいんですよね。
締め切りが早い
例: 午後5時 それ以降に入れない
締め切りが遅い
例: 午後11時 睡眠アプリとしてふさわしくない

時間で締め切る方法だと、朝グループを作った人はミッションを見るのに夜またサイトを見ないといけないという問題も考えられますね。
